src/inmanta/data/__init__.py

```diff
@@ -112,8 +112,10 @@ class TableLockMode(enum.Enum):
     Table level locks as defined in the PostgreSQL docs:
     https://www.postgresql.org/docs/13/explicit-locking.html#LOCKING-TABLES. When acquiring a lock, make sure to use the same
     locking order accross transactions to prevent deadlocks and to otherwise respect the consistency docs:
-    https://www.postgresql.org/docs/13/applevel-consistency.html#NON-SERIALIZABLE-CONSISTENCY. See relevant data classes'
-    docstrings for appropriate lock orderings.
+    https://www.postgresql.org/docs/13/applevel-consistency.html#NON-SERIALIZABLE-CONSISTENCY.
+
+    In general, locks should be acquired consistently with delete cascade lock order, which is top down. See relevant data
```

Comment:
Wouldn't it be better to have one central place where we define the total locking order. Now this information is spread across many docstring, which makes it difficult to determine the total locking order and opens the door to inconsistencies in the documentation.

Reply:
Done